fastapi/dependencies/utils.py

```diff
@@ -583,16 +591,24 @@ def request_params_to_args(
     values = {}
     errors = []
     for field in required_params:
+        field_info = field.field_info
+        assert isinstance(
+            field_info, params.Param
+        ), "Params must be subclasses of Param"
+
         if is_scalar_sequence_field(field) and isinstance(
             received_params, (QueryParams, Headers)
         ):
-            value = received_params.getlist(field.alias) or field.default
+            if isinstance(field_info, params.Query) and not field_info.explode:
+                value = received_params.get(field.alias)
+                if value is not None:
```

Comment:
You can also use [the walrus operator](https://docs.python.org/3/whatsnew/3.8.html#assignment-expressions) to merge this condition as well if the python version is compatible.

Reply:
That's true but AFAIK FastAPI tries to be compatible with python 3.6+ so I guess walrus is out of the question.